core/src/data/literal.rs

```diff
@@ -62,24 +62,48 @@ impl<'a> TryFrom<&'a AstLiteral> for Literal<'a> {
     }
 }
 
-impl PartialEq<String> for Literal<'_> {
-    fn eq(&self, other: &String) -> bool {
+impl PartialEq<str> for Literal<'_> {
+    fn eq(&self, other: &str) -> bool {
         match (self, other) {
             (Literal::Text(l), r) => l.as_ref() == r,
             _ => false,
         }
     }
 }
 
-impl PartialOrd<String> for Literal<'_> {
-    fn partial_cmp(&self, other: &String) -> Option<Ordering> {
+impl PartialOrd<str> for Literal<'_> {
+    fn partial_cmp(&self, other: &str) -> Option<Ordering> {
         match (self, other) {
             (Literal::Text(l), r) => Some(l.as_ref().cmp(r)),
             _ => None,
         }
     }
 }
 
+impl PartialEq<String> for Literal<'_> {
+    fn eq(&self, other: &String) -> bool {
+        PartialEq::<str>::eq(self, other.as_ref())
+    }
+}
+
+impl PartialOrd<String> for Literal<'_> {
+    fn partial_cmp(&self, other: &String) -> Option<Ordering> {
+        PartialOrd::<str>::partial_cmp(self, other.as_ref())
+    }
+}
+
+impl PartialEq<&str> for Literal<'_> {
+    fn eq(&self, other: &&str) -> bool {
+        PartialEq::<str>::eq(self, *other)
+    }
+}
+
+impl PartialOrd<&str> for Literal<'_> {
+    fn partial_cmp(&self, other: &&str) -> Option<Ordering> {
+        PartialOrd::<str>::partial_cmp(self, *other)
+    }
+}
```

Comment:
Can we use `AsRef<str>` here?
```rust
impl<T: AsRef<str>> PartialEq<T> for Literal<'_> {
     fn eq(&self, other: &T) -> bool {             
         PartialEq::<str>::eq(self, other.as_ref())
     }                                             
 }                                                 
```

Reply:
Oh.. `&&str` doesn't seem to be included here.

Reply:
No, I was impatient. (just before check success)